.github/workflows/generate_docs.yml

```diff
@@ -0,0 +1,29 @@
+name: Generate Terraform Plugin Documentation
```

Comment:
Correct me if I'm wrong but I think this is ephemeral? It will generate the docs but they aren't committed to the repo so any changes won't get merged. I'm thinking what we probably want here is to have this workflow compare the generated docs to what is in the repo to ensure that any updated docs have been committed?

Reply:
I have added a check to see if there are any diffs.